Patch-Source: https://github.com/flexmock/flexmock/pull/121

From ec748515e1a0f9ceb6c45e94545fd314918a9d4f Mon Sep 17 00:00:00 2001
From: Olli Paakkunainen <olli@paakkunainen.fi>
Date: Wed, 5 Jan 2022 09:33:50 +0900
Subject: [PATCH] Add python_files configuration for pytest

Pytest should only execute tests in test_pytest.py file.
This change instructs pytest to ignore any other test files.
---
 .github/workflows/coverage.yml | 2 +-
 Makefile                       | 2 +-
 pyproject.toml                 | 4 ++++
 tox.ini                        | 2 +-
 4 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/.github/workflows/coverage.yml b/.github/workflows/coverage.yml
index fe14907c..4943348f 100644
--- a/.github/workflows/coverage.yml
+++ b/.github/workflows/coverage.yml
@@ -33,7 +33,7 @@ jobs:
           poetry run coverage run --parallel-mode --source flexmock -m unittest tests/test_flexmock.py
           poetry run coverage run --parallel-mode --source flexmock tests/test_unittest.py
           poetry run coverage run --parallel-mode --source flexmock tests/test_doctest.py
-          poetry run coverage run --parallel-mode --source flexmock -m pytest tests/test_pytest.py
+          poetry run coverage run --parallel-mode --source flexmock -m pytest
           poetry run coverage combine
           poetry run coverage xml
 
diff --git a/Makefile b/Makefile
index 2a851477..3ba68aba 100644
--- a/Makefile
+++ b/Makefile
@@ -30,7 +30,7 @@ pytest:
 	@printf '\n\n*****************\n'
 	@printf '$(color)Running pytest$(off)\n'
 	@printf '*****************\n'
-	pytest tests/test_pytest.py
+	pytest
 
 .PHONY: unittest
 unittest:
diff --git a/pyproject.toml b/pyproject.toml
index 9c00e473..bd89815e 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -117,3 +117,7 @@ no_implicit_optional = true
 [[tool.mypy.overrides]]
 module = "tests.*"
 ignore_errors = true
+
+[tool.pytest.ini_options]
+python_files = ["test_pytest.py"]
+xfail_strict = true
diff --git a/tox.ini b/tox.ini
index ff2c1a76..ab8199a5 100644
--- a/tox.ini
+++ b/tox.ini
@@ -37,4 +37,4 @@ commands =
     python -m testtools.run --verbose tests/test_testtools.py
     sh -c 'python -m subunit.run tests/test_flexmock.py | subunit2pyunit'
     python tests/test_teamcity.py
-    pytest tests/test_pytest.py
+    pytest
